element-hq/element-x-android
# Refresh room summaries when date or time changes in the device

## Content

- Create `DateTimeObserver`, which will register a receiver for date and time changes in the devices and emit updates.
- Have `RoomListDataSource` listen to those updates.
- Refresh the summary timestamps when an update is received.
- Modify the DI so we have a `TimezoneProvider` instead of injecting the `Timezone` used when the app was opened.

Alternative to https://github.com/element-hq/element-x-android/pull/3650.

## Motivation and context

There is a bug where in the room list you might be able to see rooms whose state hasn't changed since yesterday still displaying the time at which some event happened instead of 'Yesterday', making you think they happened at some time today and they're out of order.

If you open one of these rooms, the room list item is also refreshed and the right 'Yesterday' formatted timestamp is used.

The same happens with the time if you change the timezone of your device.

## Screenshots / GIFs

They should be the same.

## Tests



- Open the app and let the room list load completely until it doesn't receive new updates.
- Go to system settings -> change the day. Check the timestamps change in the room list.
- Restore the current value, check it changes back to what's expected.
- Now change the timezone and check the time also changes in the room list.

## Tested devices

- OS version(s): 14

## Checklist
